jax/scipy/sparse/linalg.py

```diff
@@ -46,15 +56,15 @@ def body_fun(value):
     x, r, gamma, p, k = value
     Ap = A(p)
     alpha = gamma / _vdot(p, Ap)
-    x_ = x + alpha * p
-    r_ = r - alpha * Ap
+    x_ = _add(x, _mul(alpha, p))
```

Comment:
Actually, these games look familiar! You don't have to play them if you flatten functions ahead of time, using things like `flatten_fun`. Then you only have to deal with lists of arrays rather than arbitrary pytrees of arrays.

Reply:
Hmm. That would avoid the need for these aliases `_add` and `_mul`, but the end result wouldn't be much more readable because I still couldn't use infix arithmetic :(.